leader-election-impl/src/main/java/com/palantir/leader/LeaderElectionServiceBuilder.java

```diff
@@ -0,0 +1,145 @@
+    @Nullable private Duration pingRate;
+    @Nullable private Duration randomWaitBeforeProposingLeadership;
+    @Nullable private UUID leaderUuid;
+    private int quorumSize = -1;
+
+    public LeaderElectionServiceBuilder acceptorClient(PaxosAcceptorNetworkClient acceptorClient) {
+        this.acceptorClient = Preconditions.checkNotNull(acceptorClient, "acceptorClient cannot be null");
+        return this;
+    }
+
+    public LeaderElectionServiceBuilder learnerClient(PaxosLearnerNetworkClient learnerClient) {
+        this.learnerClient = Preconditions.checkNotNull(learnerClient, "learnerClient cannot be null");
+        return this;
+    }
+
+    public LeaderElectionServiceBuilder knowledge(PaxosLearner knowledge) {
+        this.knowledge = Preconditions.checkNotNull(knowledge, "knowledge cannot be null");
+        return this;
+    }
+
+    public LeaderElectionServiceBuilder leaderPinger(LeaderPinger leaderPinger) {
+        this.leaderPinger = Preconditions.checkNotNull(leaderPinger, "leaderPinger cannot be null");
+        return this;
+    }
+
+    public LeaderElectionServiceBuilder eventRecorder(PaxosLeaderElectionEventRecorder eventRecorder) {
+        this.eventRecorder = Preconditions.checkNotNull(eventRecorder, "eventRecorder cannot be null");
+        return this;
+    }
+
+    public LeaderElectionServiceBuilder pingRate(Duration pingRate) {
+        this.pingRate = Preconditions.checkNotNull(pingRate, "pingRate cannot be null");
+        return this;
+    }
+
+    public LeaderElectionServiceBuilder randomWaitBeforeProposingLeadership(
+            Duration randomWaitBeforeProposingLeadership) {
+        this.randomWaitBeforeProposingLeadership = Preconditions.checkNotNull(
+                randomWaitBeforeProposingLeadership, "randomWaitBeforeProposingLeadership cannot be null");
+        return this;
+    }
+
+    public LeaderElectionServiceBuilder leaderUuid(UUID leaderUuid) {
+        this.leaderUuid = Preconditions.checkNotNull(leaderUuid, "leaderUuid cannot be null");
+        return this;
+    }
+
+    public LeaderElectionServiceBuilder quorumSize(int quorumSize) {
+        this.quorumSize = Preconditions.checkNotNull(quorumSize, "quorumSize cannot be null");
+        return this;
+    }
+
+    public LeaderElectionService build() {
+        return new PaxosLeaderElectionService(
+                buildProposer(),
+                knowledge(),
+                leaderPinger(),
+                acceptorClient(),
+                learnerClient(),
+                pingRate().toMillis(),
+                randomWaitBeforeProposingLeadership().toMillis(),
+                eventRecorder());
+    }
+
+    private PaxosProposer buildProposer() {
+        return PaxosProposerImpl.newProposer(acceptorClient(), learnerClient(), quorumSize(), leaderUuid());
+    }
+
+    private int quorumSize() {
+        Preconditions.checkArgument(quorumSize > 0, "quorumSize not set");
```

Comment:
Is there a way to verify that the quorum size isn't greater than the number of nodes? I think we hide this detail now so it's not easy, in which case it's fine.

---

Also, non-actionable: There's nothing stopping someone from trying to set the quorum size in the builder to zero or a negative number.

Reply:
hmm, so it turns out, we don't actually need quorum size, because it's only there to be exposed through the `PaxosProposer`, can remove, then this comment is moot